dace/transformation/dataflow/buffer_tiling.py

```diff
@@ -0,0 +1,134 @@
+# Copyright 2019-2020 ETH Zurich and the DaCe authors. All rights reserved.
+""" Contains classes that implement the OnTheFlyMapFusion transformation. """
+
+from dace import registry
+from dace.sdfg import nodes
+from dace.sdfg import utils as sdutil
+from dace.transformation import transformation
+from dace.transformation.dataflow import MapTiling, MapTilingWithOverlap, MapFusion, TrivialMapElimination
+
+@registry.autoregister_params(singlestate=True)
+class BufferTiling(transformation.Transformation):
+    _map1_exit = nodes.MapExit(nodes.Map('', [], []))
+    _array = nodes.AccessNode('')
+    _map2_entry = nodes.MapEntry(nodes.Map('', [], []))
+
+    # Returns a list of graphs that represent the pattern
+    @staticmethod
+    def expressions():
+        return [
+            sdutil.node_path_graph(
+                BufferTiling._map1_exit,
+                BufferTiling._array,
+                BufferTiling._map2_entry,
+            )
+        ]
+
+    @staticmethod
+    def can_be_applied(graph, candidate, expr_index, sdfg, strict=False):
+        map1_exit = graph.nodes()[candidate[BufferTiling._map1_exit]]
+        map2_entry = graph.nodes()[candidate[BufferTiling._map2_entry]]
+
+        # Check that locally provided buffer has volume 1.
+        for edge in graph.in_edges(map1_exit):
+            if edge.data.volume != 1:
+                return False
+
+        # Check that all locally provided buffers have the same subset.
+        subsets = set(edge.data.subset for edge in graph.in_edges(map1_exit))
```

Comment:
same

Reply:
I removed this check. It tried to restrict BufferTiling to stencil-like programs, but I think this is no longer needed. Programs that don't satisfy this check are quite weird, but excluding them seems premature now. After all, MapTiling and MapFusion might be able to process them correctly.